modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -4683,3 +4688,150 @@ def compare(self, other, **kwargs):
                 other._modin_frame,
             )
         )
+
+    def case_when(self, caselist):
+        if Engine.get() in ("Unidist", "Dask"):
+            # https://github.com/modin-project/modin/issues/6973
+            return super().case_when(caselist)
+
+        frame = self._modin_frame
+        # If all the conditions are callable and the replacements are either
+        # callable or scalar, use map().
+        if all(
+            callable(c[0]) and (callable(c[1]) or not is_list_like(c[1]))
+            for c in caselist
+        ):
+            frame = frame.map(func=_case_when, func_args=[caselist], lazy=True)
```

Comment:
Why do we want this to be lazy? Is it more beneficial?

Reply:
There is no difference in case of a single call, but in case of multiple calls or if another lazy operation is performed on the result, it's faster because it allows to perform a batch of operations on each partition in parallel.